src/error.rs

```diff
@@ -62,6 +64,9 @@ impl fmt::Display for Error {
             Error::Io(ref e) => fmt::Display::fmt(e, f),
             Error::Tls(ref e) => fmt::Display::fmt(e, f),
             Error::TlsHandshake(ref e) => fmt::Display::fmt(e, f),
+            Error::BadResponse(ref data) => {
```

Comment:
This seems like an unrelated change?
Also, shouldn't `Error::Validate` have an entry here?

Reply:
There are 2 changes in this set: expanding BadResponse and also adding validation. Should I split them into separate PRs?

Error::Validate added in Display::fmt.